Shell: Support for ctr-e and ctr-a.

marprok opened the pull request:
By pressing ctr-e/a the cursor goes to the end/beginning of the line.

Shell: Added support for ctr-e/a.

Removed the fix from the Terminal app and added it to the Shell.
**This is a continuation of the previously closed pull request(#506).**

awesomekling commented on Shell/LineEditor.cpp:
```diff
@@ -226,6 +226,23 @@ String LineEditor::get_line(const String& prompt)
+            if (ch == 0x01) // ^A
+            {
+                if (m_cursor > 0) {
+                    fprintf(stdout, "\033[%dD", m_cursor);
```
```suggestion
                    printf("\033[%dD", m_cursor);
```

`fprintf(stdout)` is really just `printf()`

awesomekling commented on Shell/LineEditor.cpp:
```diff
@@ -226,6 +226,23 @@ String LineEditor::get_line(const String& prompt)
+            } else if (ch == 0x05) // ^E
+            {
+                if (m_cursor < m_buffer.size()) {
+                    fprintf(stdout, "\033[%dC", m_buffer.size() - m_cursor);
```
```suggestion
                    printf("\033[%dC", m_buffer.size() - m_cursor);
```

awesomekling commented on Shell/LineEditor.cpp:
```diff
@@ -226,6 +226,23 @@ String LineEditor::get_line(const String& prompt)
                 fflush(stdout);
                 continue;
             }
+            if (ch == 0x01) // ^A
```
```suggestion
            if (ch == 0x01) { // ^A
```

awesomekling commented on Shell/LineEditor.cpp:
```diff
@@ -226,6 +226,23 @@ String LineEditor::get_line(const String& prompt)
+                    m_cursor = 0;
+                }
+                continue;
+            } else if (ch == 0x05) // ^E
```
No need for `else` after a `continue`.

```suggestion
            }
            if (ch == 0x05) { // ^E
```
```

marprok commented:
Fixed the requested changes. I did use the clang format on the file, but it seems that because I had added the comments // ^A and // ^E the format did nothing. Also in some cases in the file you use fprintf and printf interchangeably so I guessed that it didn't matter what I would chose to use.

awesomekling commented:
Great! Thank you :)
